engines/scumm/scumm.cpp

```diff
@@ -338,11 +338,17 @@ ScummEngine::ScummEngine(OSystem *syst, const DetectorResult &dr)
 		break;
 
 	case Common::kRenderAmiga:
-		// Allow v2 games to be rendered in forced Amiga mode; this works, and
-		// doing this to avoid the "sunburn effect" in MM/Zak is popular.
-		// Also allow this for Indy3 EGA.
-		if (_game.platform != Common::kPlatformAmiga && _game.version != 2 &&
-(_game.version != 3 || _game.id != GID_INDY3 || _game.platform == Common::kPlatformMacintosh))
+		// Allow V2 DOS/EGA games to be rendered with the Amiga palette; this works,
+		// and doing so to avoid the "sunburnt" effect in MM/Zak is popular to the
+		// point of being suggested in some online guides. Some players also like
+		// this setting for the V3/V4 EGA titles (especially since ScummVM < 2.7.0
+		// allowed it; see Trac#15458) and one may say that LucasArts made this
+		// option "canon" if you read the opposite `Common::kRenderEGA` case above.
+		if (_game.version >= 2 && _game.version <= 4 && _game.platform == Common::kPlatformDOS &&
```

Comment:
Breaking this down.... (from [our wiki on SCUMM versions](https://wiki.scummvm.org/index.php?title=SCUMM/Versions))
- Version 2: MM V2, Zak V2
- Version 3 DOS: Indy 3 EGA, Loom EGA
- Version 4: Passport, MI1 floppy EGA, Loom VGA CD

So, all these look fine and only Loom VGA stands out, but there's an option to render EGA graphics for that version :)

Reply:
@bluegr Ah, no, Loom VGA shouldn't be allowed, my bad. I thought my initial check properly excluded the VGA variants, but there's no easy unique flag for EGA/VGA titles, it seems. I've adjusted the PR, thanks for noticing the Loom VGA case (which is neither `GF_OLD256` nor `GF_16COLOR`).

(I'm only enabling the Amiga palette for "true" EGA titles, on purpose. I don't think it will look good, or even work, for the titles offering EGA only as a dithering fallback.)